initial commit

t-900-a opened the pull request:
skeleton, not working

t-900-a committed:
initial commit

t-900-a committed:
MIT License wasn't specified correctly in manifest

t-900-a committed:
MIT License wasn't specified correctly in manifest

t-900-a committed:
black & isort ran

t-900-a committed:
black & isort ran

t-900-a committed:
initial unit test

t-900-a committed:
initial unit test

t-900-a committed:
Adding github unittest workflow

t-900-a committed:
Merge remote-tracking branch 'origin/dev' into dev

t-900-a committed:
removed pip install unittest
it is a standard module

t-900-a committed:
Remove unittest install in workflow

unittest is a standard python module

t-900-a committed:
parameters unnecessary for _get_providers()

t-900-a committed:
Merge remote-tracking branch 'origin/main' into main

t-900-a committed:
Adding coverage to unit testing run

t-900-a committed:
travisci added

t-900-a committed:
removing github workflows

t-900-a committed:
Adding OCA tools files
https://github.com/OCA/maintainer-quality-tools

t-900-a committed:
added minimum passing unit test

t-900-a committed:
added pre-commit and fixes

t-900-a committed:
using python 3.8 for travisci

t-900-a committed:
fail the unit test

t-900-a committed:
fail the unit test

t-900-a committed:
added unit test module

t-900-a committed:
pre-commit run

t-900-a committed:
pylint fixes

t-900-a committed:
flake8 fixes

t-900-a committed:
flake8 is failing, trying to fix

t-900-a committed:
removed test module

t-900-a committed:
bump back to 3.8

t-900-a committed:
adding back tests

t-900-a committed:
restructuring tests

t-900-a committed:
restructuring folders

t-900-a committed:
failing test

t-900-a committed:
hoping tests work with python 3.7 instead of 3.8

t-900-a committed:
using python 3.8, cleaned F401

t-900-a committed:
using python 3.7

t-900-a committed:
connection to database is failing during test trying odoo v 13 instead of v 14

t-900-a committed:
connection to database is failing during test trying odoo v 14 instead of v 13

t-900-a committed:
added python lxml to packages for travis-ci

t-900-a committed:
adding test database initiation

t-900-a committed:
trying to fix parse error

t-900-a committed:
trying to fix parse error

t-900-a committed:
trying to fix parse error

t-900-a committed:
working on unit tests

t-900-a committed:
changed versioning in manifest

t-900-a committed:
changed versioning in manifest

t-900-a committed:
debuging parse error

t-900-a committed:
debugging 2 unit tests

t-900-a committed:
debugging a unit tests

t-900-a committed:
including website repo

t-900-a committed:
allowing test user to create the needed db

t-900-a commented:
https://codecov.io/gh/t-900-a/moneroodoo/tree/9c04d49daaad20612c797eb1916beb6c43280679
Travis ci build now passing
https://travis-ci.com/t-900-a/moneroodoo.svg?branch=main

t-900-a committed:
updating dev branch

t-900-a committed:
Reverting to last successful travis ci state

This reverts commit 3fd73675bd3abab2b1c6325190b8687b1bc85a45.

serhack commented on monero-rpc-odoo/models/monero_acq.py:
```diff
@@ -0,0 +1,69 @@
+        help="The port the Monero RPC is listening on",
+        default=18082
+    )
+    monero_rpc_config_path = fields.Char(
```
RPC path usually never change. Better to remove that from settings.

serhack commented on monero-rpc-odoo/data/currency.xml:
```diff
@@ -0,0 +1,13 @@
+        <record id="currency_xmr" model="res.currency">
+            <field name="name">XMR</field>
+            <field name="currency_unit_label">Monero</field>
+            <field name="symbol">ɱ</field>
```
Great 👍🏻

serhack commented on monero-rpc-odoo/controllers/controllers.py:
```diff
@@ -0,0 +1,206 @@
+            "acquirer_id": payment_acquirer_id,
+            # surrogate key for payment acquirer
+            "acquirer_ref": "cryptocurrency",
+            # this should be the tx hash?
```
No! Never store something sensible as tx_hash!

t-900-a committed:
removed path configuration for monerod

t-900-a committed:
do not store tx hash
removed reference

serhack commented on monero-rpc-odoo/README.md:
```diff
@@ -0,0 +1,14 @@
+[![Build Status](https://travis-ci.com/t-900-a/moneroodoo.svg?branch=master)](https://travis-ci.com/t-900-a/moneroodoo)
+[![codecov](https://codecov.io/gh/t-900-a/moneroodoo/branch/master/graph/badge.svg)](https://codecov.io/gh/t-900-a/moneroodoo)
+
```
A suggestion for you: add CREDITS part, well done! :)

serhack commented on monero-rpc-odoo/README.md:
```diff
@@ -0,0 +1,14 @@
+
+# Monero Odoo
+
+Allows you to accept Monero as Payment within your Odoo Ecommerce shop
```
A link to Odoo project?

t-900-a committed:
add logo

t-900-a committed:
add links to readme
logo, codecov,build status

t-900-a committed:
merge
3fd73675bd3abab2b1c6325190b8687b1bc85a45

t-900-a committed:
black / isort fixes

t-900-a committed:
flake8 fixes

t-900-a committed:
added payment icon to manifest

t-900-a committed:
add monero as dependency in manifest

t-900-a committed:
add monero as external dependency in manifest

t-900-a committed:
flake8 fix

t-900-a committed:
payment acquirer record id parsing failed

t-900-a committed:
fixing test

t-900-a committed:
added log line on rpc connection failure

t-900-a committed:
more features
changed how exception handling works, the module now has it's own exceptions defined
subaddress now displays on payment confirmation page

t-900-a committed:
orders are now added to a queue

t-900-a committed:
removed scheduler.xml & added decimal_places field to currency.xml

A queue handle's transaction processing now, rather than a scheduled job

t-900-a committed:
Fixed float comparison of transaction amount

t-900-a committed:
cancel order on job max retries
if the job is on it's final retry and there is still no tx found, then cancel order

t-900-a committed:
add confirmations field / attribute / logic
now supports user defined "security levels"
low security = 0-conf
etc...

t-900-a committed:
flake8 fixes

t-900-a committed:
added sales order test

t-900-a committed:
flake8 fixes

t-900-a committed:
flake8 fixes

t-900-a committed:
fix sales order test

t-900-a committed:
fix sales order test

t-900-a committed:
flake8 fix

t-900-a committed:
trying to fix sales order unit test

t-900-a committed:
trying to fix sales order unit test

t-900-a committed:
flake8 fix

t-900-a committed:
preparing for initial release

t-900-a committed:
fixed exception when handling zero-conf transactions

t-900-a committed:
formatting changes
